From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: root <root@ROBERT-PC.localdomain>
Date: Tue, 14 Jun 2022 14:48:56 +0200
Subject: [PATCH] GERMAN PLUGINSCOMMAND

GERMAN PLUGINSCOMMAND2

GERMAN PLUGINSOMMAND2

FIX

GERMAN PLUGINS

GERMAN PLUGINS

GERMAN PLUGINS

GERMAN PLUGINS

GERMAN PLUGINS

GERMAN PLUGINS

GERMAN PLUGINS

GERMAN PLUGINS

GERMAN PLUGINS

GERMAN PLUGINS

GERMAN PLUGINS

diff --git a/src/main/java/org/bukkit/command/defaults/PluginsCommand.java b/src/main/java/org/bukkit/command/defaults/PluginsCommand.java
index 0b4937dc6d003caf0f43a01a5a0801d5804082bf..65285eae4c954ac445cadbf454145dad787980cf 100644
--- a/src/main/java/org/bukkit/command/defaults/PluginsCommand.java
+++ b/src/main/java/org/bukkit/command/defaults/PluginsCommand.java
@@ -18,12 +18,14 @@ import net.kyori.adventure.text.format.NamedTextColor;
 import net.kyori.adventure.text.event.ClickEvent;
 import org.bukkit.entity.Player;
 import org.bukkit.plugin.PluginDescriptionFile;
+import net.kyori.adventure.text.minimessage.MiniMessage; // 0xBUKKIT
+import net.kyori.adventure.text.minimessage.tag.resolver.Placeholder; // 0xBUKKIT
 // Spigot end
 
 public class PluginsCommand extends BukkitCommand {
     public PluginsCommand(@NotNull String name) {
         super(name);
-        this.description = "Gets a list of plugins running on the server";
+        this.description = "PLUGIN LISTE";
         this.usageMessage = "/plugins";
         this.setPermission("bukkit.command.plugins");
         this.setAliases(Arrays.asList("pl"));
@@ -35,9 +37,9 @@ public class PluginsCommand extends BukkitCommand {
 
         // Spigot start
         if (sender instanceof Player && sender.hasPermission("bukkit.command.version")) {
-            sender.sendMessage(getPluginListSpigot());
+            sender.sendMessage(MiniMessage.miniMessage().deserialize("<white>[<gray>Server<white>] <green>Server Plugins <gold>" + getPluginList())); // 0xBUKKIT
         } else {
-            sender.sendMessage("Plugins " + getPluginList());
+            sender.sendMessage(MiniMessage.miniMessage().deserialize("<white>[<gray>Server<white>] <green>Deine Plugins <gold>" + getPluginList())); // 0xBUKKIT
         }
         // Spigot end
         return true;
@@ -61,13 +63,12 @@ public class PluginsCommand extends BukkitCommand {
         StringBuilder pluginList = new StringBuilder();
         for (Map.Entry<String, Plugin> entry : plugins.entrySet()) {
             if (pluginList.length() > 0) {
-                pluginList.append(ChatColor.WHITE);
-                pluginList.append(", ");
+                pluginList.append(" <white>| <gold>"); // 0xBUKKIT
             }
 
             Plugin plugin = entry.getValue();
 
-            pluginList.append(plugin.isEnabled() ? ChatColor.GREEN : ChatColor.RED);
+            pluginList.append(plugin.isEnabled() ? "<green>" : "<red>"); // 0xBUKKIT
             // Paper start - Add an asterisk to legacy plugins (so admins are aware)
             String pluginName = plugin.getDescription().getName();
             if (org.bukkit.UnsafeValues.isLegacyPlugin(plugin)) {
@@ -81,7 +82,7 @@ public class PluginsCommand extends BukkitCommand {
             }
         }
 
-        return "(" + plugins.size() + "): " + pluginList.toString();
+        return "<white>[ <gold>" + plugins.size() + " <white>]<green>: <white>[ " + pluginList.toString() + " <white>]"; // 0xBUKKIT
         // Paper end
     }
 
@@ -90,12 +91,12 @@ public class PluginsCommand extends BukkitCommand {
     private TextComponent getPluginListSpigot() {
         Plugin[] plugins = Arrays.stream(Bukkit.getPluginManager().getPlugins()).sorted(java.util.Comparator.comparing(plugin -> plugin.getName().toLowerCase())).toArray(Plugin[]::new); // Purpur
         TextComponent.Builder builder = Component.text();
-        builder.append(Component.text("Plugins (" + plugins.length + "): "));
+        builder.append(MiniMessage.miniMessage().deserialize("<white>[<gray>Server<white>] <green>Deine Plugins <white>[ <gold>" + plugins.length + " <white>]<green>: ")); // 0xBUKKIT
 
         int index = 0;
         for (Plugin plugin : plugins) {
             if (index++ > 0) {
-                builder.append(Component.text(", ", NamedTextColor.WHITE));
+                builder.append(MiniMessage.miniMessage().deserialize("<white> | <gold>")); // 0xBUKKIT
             }
 
             // Event components
@@ -105,7 +106,7 @@ public class PluginsCommand extends BukkitCommand {
 
             if (description.getDescription() != null) {
                 hover.append(Component.newline())
-                    .append(Component.text("Description: ", NamedTextColor.WHITE))
+                    .append(Component.text("Beschreibung: ", NamedTextColor.WHITE)) // 0xBUKKIT
                     .append(Component.text(description.getDescription(), NamedTextColor.GREEN));
             }
 
@@ -118,9 +119,9 @@ public class PluginsCommand extends BukkitCommand {
             if (!description.getAuthors().isEmpty()) {
                 hover.append(Component.newline());
                 if (description.getAuthors().size() == 1) {
-                    hover.append(Component.text("Author: "));
+                    hover.append(Component.text("Autor: ")); // 0xBUKKIT
                 } else {
-                    hover.append(Component.text("Authors: "));
+                    hover.append(Component.text("Autoren: ")); // 0xBUKKIT
                 }
 
                 hover.append(getAuthors(description));
@@ -131,7 +132,7 @@ public class PluginsCommand extends BukkitCommand {
                 .hoverEvent(hover.build()).clickEvent(ClickEvent.suggestCommand("/version " + description.getName())));
 
             if (plugin.getDescription().getProvides().size() > 0) {
-                builder.append(Component.text(" (", NamedTextColor.WHITE)).append(Component.text(String.join(", ", plugin.getDescription().getProvides()))).append(Component.text(")"));
+                builder.append(Component.text(" (", NamedTextColor.WHITE)).append(Component.text(String.join(" | ", plugin.getDescription().getProvides()))).append(Component.text(")")); // 0xBUKKIT
             }
         }
 
@@ -145,7 +146,7 @@ public class PluginsCommand extends BukkitCommand {
 
         for (int i = 0; i < authors.size(); i++) {
             if (i > 0) {
-                builder.append(Component.text(i < authors.size() - 1 ? ", " : " and ", NamedTextColor.WHITE));
+                builder.append(Component.text(i < authors.size() - 1 ? ", " : " und ", NamedTextColor.WHITE)); // 0xBUKKIT
             }
 
             builder.append(Component.text(authors.get(i), NamedTextColor.GREEN));
